File: src/main/java/me/anton/sickcore/core/module/ModuleHandler.java
package me.anton.sickcore.core.module;

import me.anton.sickcore.modules.basic.BasicModule;
import me.anton.sickcore.modules.discord.DiscordModule;
import me.anton.sickcore.modules.motd.MOTDModule;

import java.util.Arrays;
import java.util.List;

public class ModuleHandler {

    private List<IModule> modules = Arrays.asList(new DiscordModule(), new MOTDModule(), new BasicModule());

    public void loadModules(){
        modules.forEach(IModule::load);
    }

    public void unLoadModules(){
        modules.forEach(IModule::unload);
    }

}

File: src/main/java/me/anton/sickcore/core/modules/ModuleHandler.kt
package me.anton.sickcore.core.modules

import me.anton.sickcore.core.modules.appereance.AppereanceModule
import me.anton.sickcore.core.modules.rank.RankModule

class ModuleHandler {

    val modules = listOf(RankModule(), AppereanceModule())

    fun start(){
        modules.forEach {
            it.start()
        }
    }

    fun shutdown(){
        modules.forEach {
            it.shutdown()
        }
    }

}